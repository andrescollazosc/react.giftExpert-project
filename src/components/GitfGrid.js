import React from "react";
import { useFetchGifts } from "../hooks/useFetchGifts";
// import { getGift } from "../helpers/getGifts";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GitfGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data, loading } = useFetchGifts(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading ?? <p className="animate__animated animate__fadeIn">Loading</p>}

      <div className="card-grid">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GitfGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
