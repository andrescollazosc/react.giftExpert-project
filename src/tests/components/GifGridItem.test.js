import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Testing <GifGridItem /> component", () => {
  const title = "This is a title";
  const url = "https://mundoloco.com/api/human/homer.gif";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("should return text empty", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should be have a <p></p> with a text.", () => {
    const p = wrapper.find("p").text().trim();
    expect(p).toBe(title);
  });

  test("should be have src img equal to url of the props", () => {
    const img = wrapper.find("img").props();

    expect(img.src).toBe(url);
    expect(img.alt).toBe(title);
  });

  test('Should be hace animate__fadeIn', () => {
      const div = wrapper.find('div');
      const className = div.prop('className');
      

      expect(className.includes('animate__fadeIn')).toBe(true);
  });
});
