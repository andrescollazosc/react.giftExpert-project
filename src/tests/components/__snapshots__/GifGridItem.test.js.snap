// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGridItem /> component should return text empty 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="This is a title"
    src="https://mundoloco.com/api/human/homer.gif"
  />
  <p>
    This is a title
  </p>
</div>
`;
